util/mcf_dict_util.py

```diff
@@ -0,0 +1,537 @@
+        data_["dcids"] = dcid_chunk
+        req = request_post_json(
+            f'https://{url_prefix}api.datacommons.org/node/property-labels',
+            data_)
+        resp_dicts = req['payload']
+        resp_dicts = ast.literal_eval(resp_dicts)
+        for cur_dcid in resp_dicts:
+            if not resp_dicts[cur_dcid]:
+                ret_dict[cur_dcid] = False
+            elif not resp_dicts[cur_dcid]['inLabels'] and not resp_dicts[
+                    cur_dcid]['outLabels']:
+                ret_dict[cur_dcid] = False
+            else:
+                ret_dict[cur_dcid] = True
+
+    return ret_dict
+
+
+def mcf_to_dict_list(mcf_str: str) -> list:
+    """Convets MCF file string to a list of OrderedDict objects.
+
+    Args:
+        mcf_str: String read from MCF file.
+        
+    Returns:
+        List of OrderedDict objects where each object represents a node in the MCF file.
+    """
+    # TODO preserve empty lines if required
+    # split by \n\n
+    nodes_str_list = mcf_str.split('\n\n')
+    # each node
+    node_list = []
+
+    for node in nodes_str_list:
+        node = node.strip()
+        # check for comments
+        node_str_list = node.split('\n')
+        cur_node = OrderedDict()
+        comment_ctr = 0
+        is_first_prop = True
+        # add each pv to ordered dict
+        for pv_str in node_str_list:
+            # TODO handle multiple occurrences of same property within a node
+            pv_str = pv_str.strip()
+            if pv_str.startswith('#'):
+                cur_node[f'__comment{comment_ctr}'] = pv_str
+                comment_ctr += 1
+            elif is_first_prop:
+                is_first_prop = False
+                if pv_str and not pv_str.startswith('Node: '):
+                    raise ValueError('Each node must start with Node: <name>".')
+            if pv_str and not pv_str.startswith('#'):
+                # find p, prefix, v
+                pv = pv_str.split(':')
+                if pv_str.count(':') == 1:
+                    p = pv[0].strip()
+                    prefix = ''
+                    v = pv[1].strip()
+                elif pv_str.count(':') == 2:
+                    p = pv[0].strip()
+                    prefix = pv[1].strip()
+                    v = pv[2].strip()
+                else:
+                    p = pv[0].strip()
+                    prefix = pv[1].strip()
+                    v = ':'.join(pv[2:]).strip()
+                    # TODO detect colon within a str(for e.g. descriptionURL)
+                    print(
+                        f"Warning - unexpected number of ':' found in {pv_str}")
+
+                # TODO (optional) if not prefix
+                # if not complex value and p != 'dcid'
+                # prefix = 'dcs'
+                cur_node[p] = {}
+                cur_node[p]['value'] = v
+                if v.startswith('[') and v.endswith(']'):
+                    cur_node[p]['complexValue'] = re.sub(' +', ' ',
```

Comment:
Please describe this block in a comment. It seems specific to us census to remove `+`?